uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -250,25 +274,45 @@ define([
         vm.error = undefined;
       };
 
-      (function () {
-        // Create an empty leave request
-        vm.leaveRequest = LeaveRequestInstance.init({
-          contact_id: contactId //resolved from directive
-        }, false);
+      /**
+       * initializes the controller on loading the dialog
+       */
+      (function initController() {
+        if (directiveOptions.leaveRequest) {
+          //manager is responding to leave request
+          vm.uiOptions.isManager = directiveOptions.leaveRequest.contact_id != directiveOptions.contactId;
```

Comment:
I don't find this clear

> 1. First isManager is set based on the id's coming from directive

But it's set based on *wrong* logic

> 2. and then it can be reset in isManagedBy call if given manager is not of staff

Why does it need to be reset at all? Why can't you simply set `isManager` and `isOwner` in one go, in a single function?